### PR TITLE
Sticky Notes 2.1.0

### PR DESCRIPTION
**Pull request by Fjorge0**
-Add auto launch sticky note on load setting
-Add settings
-Change some functions to use goosemodScope
-Move launching initial note into `onLoadingFinished()` function
-Delete comment block with button SVG code
-Various other tweaks

### CODE REVIEW

File: modules/ui/stickyNotes.js [hunk 1]
```diff
@@ -1,10 +1,24 @@
-const version = '2.0.1';
+const version = '2.1.0';
 
 let newStickyKeybindFunction;
 
 function injMod() {
+
+    goosemodScope.stickyNotes = {
```
Comment (CanadaHonk) [inline modules/ui/stickyNotes.js]: Do not pollute the global scope like this, why was this changed / used? Just use variables in the module.

**Review by CanadaHonk — CHANGES_REQUESTED** [review]

File: modules/ui/stickyNotes.js (hunk 1 above)
Comment (Fjorge0) [reply modules/ui/stickyNotes.js]: I more or less just followed the Visual Tweaks module as an example of how to make settings work. I also want them to save and it looked like Visual Tweaks knew how to do that.

Comment (Fjorge0) [reply modules/ui/stickyNotes.js]: And for the reason I want it to save, there's no point setting the setting if it won't save across loads, because it's for something that happens on module load.

Comment (CanadaHonk) [reply modules/ui/stickyNotes.js]: Visual tweaks is very old so probably shouldn't be followed at the moment. I'd recommend just using variables instead of modifying the GooseMod scope.

Comment (CanadaHonk) [reply modules/ui/stickyNotes.js]: > And for the reason I want it to save, there's no point setting the setting if it won't save across loads, because it's for something that happens on module load.

What? With save settings and load settings you can just set a variable as the setting like in other modules.

Comment (Fjorge0) [reply modules/ui/stickyNotes.js]: Ah, well idk how much of the module things work because theres very little documentation on it, but good to know.

Comment (CanadaHonk) [reply modules/ui/stickyNotes.js]: > Ah, well idk how much of the module things work because theres very little documentation on it, but good to know.

Fair enough.

Comment (Fjorge0) [reply modules/ui/stickyNotes.js]: Could you point me to an example of a module which does this well?

Comment (CanadaHonk) [reply modules/ui/stickyNotes.js]: > Could you point me to an example of a module which does this well?

In terms of settings? Macros is a decent example.

Settings is pretty simple.

(I forgot the exact names of functions as I'm on mobile.)

In the first function you give it settings to store as an array.
In the second it gives you the array you last gave it.

If you define a variable top level (at the start) and then give it to the function which gets the settings, it will save it.

Then when loading it, in the arguments load it as something named similar and then set the variable you want to it.

JS example (not using object structure but generic):
```javascript
let thingToSave = 'thing!';

const getSettings = () => [ thingToSave ];
const loadSettings = ([ _thingToSave ]) => {
  thingToSave = _thingToSave;
};
```

Comment (Fjorge0) [reply modules/ui/stickyNotes.js]: and this will save the settings too?

Comment (CanadaHonk) [reply modules/ui/stickyNotes.js]: > and this will save the settings too?

That code isn't direct but that's how the two handler functions act and how most modules use it.

Comment (Fjorge0) [reply modules/ui/stickyNotes.js]: alright then.

**Review by CanadaHonk — APPROVED** [review]
Looks good, thanks for the changes.

**Comment by Fjorge0** [comment]
Thanks for the feedback
